derp/derp_client.go

```diff
@@ -347,12 +347,20 @@ type ReceivedPacket struct {
 
 func (ReceivedPacket) msg() {}
 
-// PeerGoneMessage is a ReceivedMessage that indicates that the client
-// identified by the underlying public key had previously sent you a
-// packet but has now disconnected from the server.
-type PeerGoneMessage key.NodePublic
+// PeerGoneMessageDisconnected is a ReceivedMessage that indicates
+// that the client identified by the underlying public key had
+// previously sent you a packet but has now disconnected from the
+// server.
+type PeerGoneMessageDisconnected key.NodePublic
 
-func (PeerGoneMessage) msg() {}
+func (PeerGoneMessageDisconnected) msg() {}
+
+// PeerGoneMessageUnknown is a ReceivedMessage that indicates that the
+// client identified by the underlying public key is not connected to
+// this server and we don't know why.
+type PeerGoneMessageUnknown key.NodePublic
```

Comment:
Here `Unknown` means the server doesn't know the reason, but also *could* be used to mean the reason code was a value that the client doesn't know. Maybe we can disambiguate those cases more explicitly in the naming of things. Relates to classification of unknown reason code values below.

Reply:
Thank you, this is now one PeerGoneMessage and the reasons are named more clearly.